Application/Education/Certificate/Generator/Service/DataCertificate/SDataPrimary.php

```diff
@@ -0,0 +1,197 @@
+<?php
+namespace SPHERE\Application\Education\Certificate\Generator\Service\DataCertificate;
+
+use SPHERE\Application\Education\Certificate\Generator\Service\Data;
+use SPHERE\Application\Education\Lesson\Division\Division;
+
+class SDataPrimary
+{
+    public static function setCertificateStandard(Data $Data)
+    {
+
+        self::setGsHjOneInfo($Data);
+        self::setGsHjInformation($Data);
+        self::setGsJOne($Data);
+        self::setGsJa($Data);
+        self::setBeGs($Data);
+    }
+
+    /**
+     * @param Data $Data
+     */
+    private static function setGsHjOneInfo(Data $Data)
+    {
+
+        $tblCertificate = $Data->createCertificate('Grundschule Halbjahresinformation', 'der ersten Klasse',
+            'GsHjOneInfo');
+        if ($tblCertificate) {
+            if ($Data->getTblSchoolTypePrimary()) {
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeHalfYear(), $Data->getTblSchoolTypePrimary(), null, true);
+                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)) {
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '1'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+            // Begrenzung des Bemerkungsfeld
+            // erste Klasse nicht, wegen Enter
+//            $FieldName = 'Remark';
+//            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+//                $Data->createCertificateField($tblCertificate, $FieldName, 4000);
+//            }
+        }
+    }
+
+    /**
+     * @param Data $Data
+     */
+    private static function setGsJOne(Data $Data)
+    {
+
+        $tblCertificate = $Data->createCertificate('Grundschule Jahreszeugnis', 'der ersten Klasse', 'GsJOne');
+        if ($tblCertificate) {
+            if ($Data->getTblSchoolTypePrimary()) {
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeYear(), $Data->getTblSchoolTypePrimary());
+                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)) {
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '1'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+            // Begrenzung des Bemerkungsfeld
+            // erste Klasse nicht, wegen Enter
+//            $FieldName = 'Remark';
+//            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+//                $Data->createCertificateField($tblCertificate, $FieldName, 4000);
+//            }
+        }
+    }
+
+    /**
+     * @param Data $Data
+     */
+    private static function setGsHjInformation(Data $Data)
```

Comment:
Codacy found an issue: [The method setGsHjInformation() has a Cyclomatic Complexity of 12. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)